### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by getroot** [AirenSoft/OvenMediaEngine#548]
OvenMediaEngine's development roadmap will be managed in this issue.

This does not mean that they are developed in this order. Because the person in charge of development for each part is different, and they cannot develop OvenMediaEnigne when they are busy.

And the priority may be changed in the future due to various circumstances.

This roadmap will continue to be updated.

[Highest and Endless]
- Higher stability
- Performance improvement
- Compatibility improvement with other systems

[High]
- Reload settings without restart 
- ABR (WebRTC, HLS) #132, #460 
- HTTP keep-alive #279 
- Low-Latency HLS #480 

[Medium]
- Advanced API function (stream deletion, etc.) #401 
- Implement RTSP features (Basic auth, pass query option...)  #443, #420
- RTMP Pull #42 

[Low]
- Implementation of WebRTC specification that has not yet been developed
- Add CORS in API Server #524 
- RTMPS Provider / Push Publisher #315 
- Support VP9, AV1
- Support SVC
- Support OPUS over SRT (after standardization)

**Comment by getroot** [comment]
We are discussing giving up LLDASH and focusing on LLHLS instead. Because LLDASH players have taken so much of our time, and we think they will continue to do so.

**Comment by bchah** [comment]
Thank you @getroot this gives us a great breakdown of where the priorities are 😀

Much gratitude to you and the entire Airensoft team for giving us these amazing tools.

**Comment by llspalex** [comment]
Thank you @getroot! 

> Please let me know if I've missed anything previously discussed.

Scaling of origin servers was discussed earlier this year with some interesting insights: https://github.com/AirenSoft/OvenMediaEngine/issues/343#issuecomment-882179919

**Comment by nums** [comment]
Thanks you @getroot  for all these informations.
Features in high priority is exactly what I'm waiting for :)
You didn't mention recording and push publishing wich still in "beta". It's maybe part of Higher stability ?

**Comment by getroot** [comment]
@llspalex 
Thanks for reminding me of my memories.
It has not yet been included in the roadmap as a way to scale the origin server has not been designed yet. I'll keep thinking about it when I have time, whether it requires an external server to collect and serve the list of origins and streams, or if there is a better way.

**Comment by getroot** [comment]
@nums 
Yes, those functions will be verified while being used in a real environment, and the beta tag will be removed after they are fully used. It's included in "Higher stability" or "Compatibility improvement with other systems" as you say.

**Comment by neodavids** [comment]
How about Adaptive Bitrate support in HLS stream? That would be game changing!

**Comment by getroot** [comment]
@neodavids The [High] priority list already has that task.

**Comment by getroot** [comment]
Support for HTTP/1.1 Persistent Connection is complete. For more details, please check the task. (https://github.com/AirenSoft/OvenMediaEngine/issues/279#issuecomment-1075284675) Thanks a lot for testing and reporting bugs!

**Comment by leporel** [comment]
I would like VP9 to be supported not only by the GPU, but also by the CPU

**Comment by sinnrrr** [comment]
> Please let me know if I've missed anything previously discussed.

please, see this https://github.com/AirenSoft/OvenMediaEngine/issues/330
thanks for a great work!

**Comment by getroot** [comment]
LLHLS released in pre-alpha. Please discuss further issues related to LLHLS in the issues below. Thank you!

https://github.com/AirenSoft/OvenMediaEngine/issues/766

**Comment by getroot** [comment]
My next task is ABR. I will apply ABR to LLHLS first, then WebRTC.

Since LLDASH and DASH will be deprecated in the future, I will not make any effort to develop ABR on LLDASH and DASH.

**Comment by getroot** [comment]
As we all know, the most important tasks, LLHLS and ABR, have been completed. I will be restructuring the roadmap and re-prioritizing it soon. I look forward to hearing from you on our next roadmap!

**Comment by getroot** [comment]
https://github.com/AirenSoft/OvenMediaEngine/issues/112 IPv6 support will be a high priority.

**Comment by getroot** [comment]
This issue will be closed and a new 2023 Roadmap issue will be created.
